### ISSUE TITLE
Bug: [sdk list <candidate> and install not working]

### ISSUE THREAD

**Issue by AJ8GH** [sdkman/sdkman-cli#1264]
**Bug report**

sdk list <candidate> is not working for any candidates, an empty list is displayed. e.g. for `sdk list java`:
```
~
~
~
~
~
~
~
(END)
```

In addition, the install command is not working at all for any candidate or any version. When running `sdk install <candidate>` or `sdk install <candidate> <version>`, e.g. `sdk install java` or `sdk install java 17.0.9-amzn`  I get the following:
```
Stop! java 17.0.9-amzn is not available. Possible causes:
 * 17.0.9-amzn is an invalid version
 * java binaries are incompatible with your platform
 * java has not been released yet

...skipping...

Tip: see all available versions for your platform:

  $ sdk list java
```

sdkman has previously worked on this machine without issue for a couple of years, I've been able to list and install candidates without problems.

So far I have tried uninstalling and reinstalling sdkman, which hasn't worked.

Currently upgrading macOS to `14.2.1` to see if it helps

**To reproduce**

install sdkman

run `sdk list java`
run `sdk install java`

**System info**

* OS: `MacOS Sonoma Version 14.0 (23A344)`
* Chip: `M1 chip`
* Shell: `zsh 5.9 (x86_64-apple-darwin23.0)`
* `sdk version` output:
```
SDKMAN!
script: 5.18.2
native: 0.4.3
```

**Comment by helpermethod** [comment]
Hi @AJ8GH!

Do you have any aliases defined which override common commands, like `curl`, `zip` etc.?

**Comment by AJ8GH** [comment]
Hey @helpermethod, no, I've got no aliases overriding those commands. the only overriding aliases I have are:
```
less=cless
ll='exa -al -h'
lm='exa -alt modified'
ls='ls -G'
mv='mv -I'
```

**Comment by AJ8GH** [comment]
I've upgraded MacOS to 14.2.1 now which hasn't helped

**Comment by helpermethod** [comment]
Did you install SKDMAN! by exactly following the installation instructions, i.e.

    curl -s "https://get.sdkman.io" | bash

There is currently an issue when piping the output of `curl -s "https://get.sdkman.io"` to zsh.

**Comment by AJ8GH** [comment]
Yes I did use the same command with the pipe to bash. I have just tried uninstalling and reinstalling again with this command to make sure, and the problem persists.

I'm wondering if it's an issue with my shell config, I'm going to try reinstalling zsh.

Please let me know if you have any suggestions.

**Comment by AJ8GH** [comment]
@helpermethod, I've solved it!

I uninstalled and then reinstalled oh-my-zsh. This alone didn't fix it, but it reminded me that I had an alias `cat='ccat'`. This was breaking sdkman, now I've removed it, it works fine.

In case this is a bug that needs looking into, ccat is a command from the zsh colorize plugin https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/colorize/README.md

I also have an alias `less='cless' overriding less with the colorize version, but this works fine and doesn't break sdk man.

Thanks for your help with this issue.
